app/models/course/forum.rb

```diff
@@ -6,6 +6,48 @@ class Course::Forum < ActiveRecord::Base
   has_many :topics, dependent: :destroy, inverse_of: :forum
   has_many :subscriptions, dependent: :destroy, inverse_of: :forum
 
+  # @!method self.with_topic_count
+  #   Augments all returned records with the number of topics in that forum.
+  scope :with_topic_count, (lambda do
+    joins { topics.outer }.
+      select { count(topics.id).as(topic_count) }.
+      group { course_forums.id }
+  end)
+
+  # @!method self.with_topic_post_count
+  #   Augments all returned records with the number of topic posts in that forum.
+  scope :with_topic_post_count, (lambda do
+    joins { topics.outer }.
+      joins("LEFT OUTER JOIN course_discussion_topics cdt on
```

Comment:
I'm sure this SQL can get correct results, but I just don't know how to translate this using squeel. Can you provide some suggestions? @lowjoel